libminifi/src/core/ProcessGroup.cpp

```diff
@@ -156,6 +168,49 @@ void ProcessGroup::startProcessing(const std::shared_ptr<TimerDrivenSchedulingAg
           break;
       }
     }
+    catch (const std::exception &e) {
+      logger_->log_error("Failed to start processor %s (%s): %s", processor->getUUIDStr(), processor->getName(), e.what());
+      failed_processors.insert(processor);
+    }
+    catch (...) {
+      logger_->log_error("Failed to start processor %s (%s)", processor->getUUIDStr(), processor->getName());
+      failed_processors.insert(processor);
+    }
+  }
+  failed_processors_ = std::move(failed_processors);
+
+  for (auto& processor : failed_processors_) {
+    try {
+      processor->onUnSchedule();
```

Comment:
why not call notifystop here?

Reply:
Default implementation of onUnSchedule calls notifyStop, but the you are free to override that in your processor.
I think this matches the way it's done in NiFi.

Reply:
I agree that by default this should be notifyStop, but I like that it is overrideable by the implementation, because it leaves space for smarter failure-handling logics, so I am OK with this as it is.